### ISSUE TITLE
Unit tests failing in fixes and develop branches

### ISSUE THREAD

**Issue by jamesmyatt** [eclipse/paho.mqtt.python#327]
Unit tests are failing on the two main development branches (https://travis-ci.org/eclipse/paho.mqtt.python/branches). This makes it hard to unit test new submissions.

**Comment by PierreF** [comment]
I've pushed a commit on develop branch that should avoid the connection reset error that happen from time to time (nearly every time on Travis for last tests :().

I was able to reproduce it on my laptop by:
* Forcing the CPU frequency to the slowest: `cpupower frequency-set -u 400Mhz`
* Having other process consuming CPU (like Firefox, at 4Ghz my laptop use 20-30% of CPU, at 400Mhz is 100%)
* Run the test

The result is a connection reset (but still not 100% reproducible. It seems the more the CPU is used the more like to produce the error). Using wireshark I've seen the following:
![capture d ecran de 2018-09-02 11-09-45](https://user-images.githubusercontent.com/359833/44954327-bfc29180-aea0-11e8-860d-754eaabfa470.png)

Which show that client does:
* Send the publish message
* Send the reset (result of sock.close() I din't) *before* broker had time to ack the message

I know that closing TCP connection while some data may be pending (the ack message from broker) could cause reset. It seems we are in this case.

As a short term fix, I've added a simple sleep in the test code.
